pylearn2/models/mlp.py

```diff
@@ -3246,9 +3301,19 @@ def fprop(self, state_below):
                                                        "pooling.")
 
             if self.pool_type == 'max':
-                p = max_pool(bc01=d, pool_shape=self.pool_shape,
-                             pool_stride=self.pool_stride,
-                             image_shape=self.detector_space.shape)
+                # TODO Can they be merged?
+                if cudnn2d:
+                    # CPU implementation
+                    # TODO check if this can be ported to the GPU one
+                    # TODO check axes: should they be B01C or BC01 or will this
+                    # always work?
+                    p = downsample.max_pool_2d(d, self.pool_shape,
```

Comment:
You are not passing `self.pool_stride` here, which is why the dimensions are not right.